File: my_flutter_gen/lib/feature/auth/ui/login/login_view_model.dart
import 'package:flutter/foundation.dart';
import 'package:my_flutter_gen/core/network/api_result.dart';
import 'package:my_flutter_gen/feature/auth/domain/usecase/login_usecase.dart';
import 'package:my_flutter_gen/feature/auth/ui/login/login_ui_state.dart';

/// 登录页 ViewModel（ChangeNotifier）
class LoginViewModel extends ChangeNotifier {
  final LoginUseCase _loginUseCase;

  LoginViewModel({required LoginUseCase loginUseCase})
      : _loginUseCase = loginUseCase;

  LoginUiState _state = LoginUiState.initial;
  LoginUiState get state => _state;

  /// 执行登录
  Future<void> login({
    required String email,
    required String password,
  }) async {
    _state = _state.copyWith(isLoading: true, errorMessage: null);
    notifyListeners();

    final result = await _loginUseCase(email: email, password: password);

    _state = result.when(
      success: (user) => _state.copyWith(isLoading: false, user: user),
      failure: (error) =>
          _state.copyWith(isLoading: false, errorMessage: error.message),
    );
    notifyListeners();
  }

  /// 清除错误信息
  void clearError() {
    _state = _state.copyWith(errorMessage: null);
    notifyListeners();
  }
}

File: MyComposeProject/app/src/main/java/com/louis/mycomposeproject/presentation/login/LoginViewModel.kt
package com.louis.mycomposeproject.presentation.login

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.louis.mycomposeproject.domain.usecase.LoginUseCase
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch

data class LoginState(
    val isLoading: Boolean = false,
    val error: String? = null,
    val isSuccess: Boolean = false
)

class LoginViewModel(private val loginUseCase: LoginUseCase) : ViewModel() {
    private val _state = MutableStateFlow(LoginState())
    val state = _state.asStateFlow()
    
    fun login(email: String, password: String) {
        viewModelScope.launch {
            _state.value = LoginState(isLoading = true)
            loginUseCase(email, password)
                .onSuccess {
                    _state.value = LoginState(isSuccess = true)
                }
                .onFailure {
                    _state.value = LoginState(error = it.message)
                }
        }
    }
}
